server/common/config/base_config.py

```diff
@@ -0,0 +1,104 @@
+import copy
+
+from flatten_dict import flatten
+from server.common.errors import ConfigurationError
+
+
+class BaseConfig(object):
+    """This class handles the mechanics of updating and checking attributes.
+    Derived classes are expected to store the actual attributes"""
+
+    def __init__(self, app_config, default_config, dictval_cases={}):
+        # reference back to the app_config
```

Comment:
Can you explain here what the relationship of BaseConfig and AppConfig is here? AppConfig contains BaseConfig?

Reply:
AppConfig contains one or more DatasetConfig and one ServerConfig.   DatasetConfig and ServerConfig both inherit from BaseConfig.

Reply:
discussed offline